### DISCUSSION TITLE
LFX Mentorship 2023 02-Jun-Aug Challenge - for #2470

### DISCUSSION THREAD

**Discussion by juntao** [WasmEdge/WasmEdge#2525] [Show and tell]
# Introduction

For those who want to apply for LFX mentorship for https://github.com/WasmEdge/WasmEdge/issues/2470, this is a pretest for the application. The goal of this LFX mentorship is to create a stream log processing framework for WasmEdge. We designed this challenge to evaluate the candidates.

# Requirements

1 [Install WasmEdge with the Rust toolchain](https://wasmedge.org/docs/develop/rust/setup)

2 [Try this example of a log processor that reads from MySQL binlogs and sends the logs to a Kafka queue](https://github.com/second-state/wasmedge-mysql-binlog-kafka)

3 Please modify it so that

3.1 It only processes binlogs for certain tables from the MySQL database. The table names can be passed in as an environment variable or be hardcoded in a Vec in the code. 

3.2 It sends binlogs from different tables into different topics in the Kafka queue. The queue naming convention is $databaseName_$tableName.  

3.3 GitHub Actions to demonstrate the result.

4 Once you are finished, please provide your link to the GitHub repository as a comment under this discussion, which must contain a README file to show how you do it.

# Deadline

Please refer to [the timeline of LFX mentorship 2023 02-Jun-Aug](https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2023/02-Jun-Aug).
Since we have to process the internal review and decide, the final date for valid submissions will be May 27, 5:00 PM PDT.

**Comment by carrycooldude** [comment]
I am facing this issue on wasedge_wasi_socket @juntao https://github.com/second-state/wasmedge_wasi_socket/issues/85

**Comment by Vaishnavi2445** [comment]
I have a question about where I can submit my GitHub repository link.  Could you please let me know the appropriate platform or method to submit the GitHub repository link?

**Reply by alabulei1** [reply]
J

> I have a question about where I can submit my GitHub repository link. Could you please let me know the appropriate platform or method to submit the GitHub repository link?

You can just comment under this discussion.

**Reply by Vaishnavi2445** [reply]
okay! Thank You so much.

**Reply by Vaishnavi2445** [reply]
Hello !
I am pleased to inform you that the project has been successfully completed. You can access the project files and view the final outcome by clicking on the following link :  https://github.com/Vaishnavi2445/wasmedge-mysql-binlog-kafka.
Thank you !

**Comment by markcty** [comment]
Hi! I've completed the challenge. Here is the link: https://github.com/markcty/wasmedge-mysql-binlog-kafka. Thanks for reviewing it!

**Comment by Priyansh61** [comment]
Hi @juntao @alabulei1 , 

I had a great time completing the challange and really got to learn so many things . 

Link : https://github.com/Priyansh61/wasmedge-mysql-binlog-kafka

Thanks,

**Comment by h4shk4t** [comment]
Hello @juntao @alabulei1 

I have completed the tasks and had a really fun time learning all sorts of new things. I hope that the quality of my work is upto your standards.

Link: https://github.com/h4shk4t/wasmedge-mysql-binlog-kafka

Best Regards
Ashutosh

**Reply by h4shk4t** [reply]
I have added another test for the database but I am unable to compile it to wasm successfully, perhaps if someone could help me with that I could demonstrate the tests of all queries in my code.

**Comment by ArberSephirotheca** [comment]
Hope its not too late.
https://github.com/ArberSephirotheca/wasmedge-mysql-binlog-kafka
I was having fun play around with this small project, a bit frustrated when dealing with wasi cross compilation thu.

**Reply by h4shk4t** [reply]
Hello, out of curiosity how did you solve the wasi cross compilation errors if you don't mind sharing? I was running into similar errors while trying to compile SQL tests.
